src/DataTable.php

```diff
@@ -379,9 +387,24 @@ public function loadData(): void
     {
         $this->initialized = true;
 
-        $baseQuery = $this->buildSearch();
-        $query = $baseQuery->clone();
+        $query = $this->buildSearch();
+        $baseQuery = $query->clone();
+
+        $result = $this->getResultFromQuery($query);
+
+        if ($aggregates = $this->getAggregate($baseQuery)) {
+            $result['aggregates'] = ! $this->search ? $aggregates : [];
```

Comment:
What if $result is not an array?

Reply:
moved below setData and set to $this->data